src/backend/joanie/core/models/products.py

```diff
@@ -383,20 +383,28 @@ def find_installments(self, due_date):
     def find_today_installments(self):
         """Retrieve orders with a payment due today."""
         due_date = timezone.now().date().isoformat()
-        return (
+        orders = (
             super()
             .get_queryset()
             .filter(
                 state__in=[
                     enums.ORDER_STATE_PENDING,
                     enums.ORDER_STATE_PENDING_PAYMENT,
                 ],
-                payment_schedule__contains=[
-                    {"due_date": due_date, "state": enums.PAYMENT_STATE_PENDING}
-                ],
+                payment_schedule__contains=[{"state": enums.PAYMENT_STATE_PENDING}],
             )
         )
 
+        return [
+            order
+            for order in orders
+            if any(
+                installment["state"] == enums.PAYMENT_STATE_PENDING
+                and installment["due_date"] <= due_date
+                for installment in order.payment_schedule
+            )
+        ]
```

Comment:
We don't return a queryset anymore, which is unexpected for a models.Manager

Maybe we could rename the function `find_pending_installments` and filter out the future due_dates before launching a celery task ?